src-plugins/itkProcessRegistrationDiffeomorphicDemons/itkProcessRegistrationDiffeomorphicDemons.cpp

```diff
@@ -117,24 +117,53 @@ QString itkProcessRegistrationDiffeomorphicDemons::identifier() const
 
 
 template <typename PixelType>
-        int itkProcessRegistrationDiffeomorphicDemonsPrivate::update()
+int itkProcessRegistrationDiffeomorphicDemonsPrivate::update()
 {
     typedef itk::Image< PixelType, 3 >  FixedImageType;
     typedef itk::Image< PixelType, 3 >  MovingImageType;
 
     //unfortunately diffeomorphic demons only work with double or float types...
     // so we need to use a cast filter.
     typedef itk::Image< float, 3 > RegImageType;
+    typedef itk::CastImageFilter< FixedImageType, RegImageType > FixedCastFilterType;
+    typedef itk::CastImageFilter< MovingImageType, RegImageType > MovingCastFilterType;
+    typename FixedCastFilterType::Pointer fixedCastFilter = FixedCastFilterType::New();
+    fixedCastFilter->SetInput((FixedImageType*)proc->fixedImage().GetPointer());
+    fixedCastFilter->Update();
+    typename MovingCastFilterType::Pointer movingCastFilter = MovingCastFilterType::New();
+    movingCastFilter->SetInput((MovingImageType*)proc->movingImages()[0].GetPointer());
+    movingCastFilter->Update();
+
+    // Spacing needs to be the same in both dataset
     typedef double TransformScalarType;
+    typedef itk::ResampleImageFilter< RegImageType,RegImageType,TransformScalarType > ResampleFilterType;
```

Comment:
I agree but at the same time, the resampling was already done below, so it was a bit the regular behavior i think. @fcollot what do you think?